### PR TITLE
Adds a check_tick to pneumatic cannon examine

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

**Comment by Tacolizard** [comment]
what does CHECK_TICK do though?

**Comment by Iamgoofball** [comment]
stops lag

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Do not print while lag checking. Build a list of strings to show to the user and show it all at once when it's built

**Comment by kevinz000** [comment]
done

File: code/game/objects/items/weapons/pneumaticCannon.dm [hunk 1]
```diff
@@ -32,14 +32,16 @@
 
 /obj/item/weapon/pneumatic_cannon/examine(mob/user)
 	..()
+	var/text = ""
```
Comment (lzimann) [inline code/game/objects/items/weapons/pneumaticCannon.dm]: var/list/text = list()

File: code/game/objects/items/weapons/pneumaticCannon.dm [hunk 2]
```diff
@@ -32,14 +32,16 @@
...
 		return
 	for(var/obj/item/I in loadedItems)
-		to_chat(user, "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>")
+		text += "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>"
+		CHECK_TICK
 	if(tank)
-		to_chat(user, "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>")
-
+		text += "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>"
+	to_chat(user, text)
```
Comment (lzimann) [inline code/game/objects/items/weapons/pneumaticCannon.dm]: `text.Join()`

File: code/game/objects/items/weapons/pneumaticCannon.dm [hunk 3]
```diff
@@ -32,14 +32,16 @@
 
 /obj/item/weapon/pneumatic_cannon/examine(mob/user)
 	..()
+	var/list/text = ""
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/pneumaticCannon.dm]: ```= list()```

File: code/game/objects/items/weapons/pneumaticCannon.dm [hunk 4]
```diff
@@ -32,14 +32,16 @@
...
 		return
 	for(var/obj/item/I in loadedItems)
-		to_chat(user, "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>")
+		text += "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>"
+		CHECK_TICK
 	if(tank)
-		to_chat(user, "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>")
-
+		text += "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>"
+	to_chat(user, text.Join())
```
Comment (lzimann) [inline code/game/objects/items/weapons/pneumaticCannon.dm]: I am not exactly sure why this line is erroring. Try doing a different variable name.
Also, forgot the first time, do `text.Join("<br>")` to merge the list with line endings.

Comment (kevinz000) [inline code/game/objects/items/weapons/pneumaticCannon.dm]: woops.

**Pull request merged**